Packs/Thycotic/Integrations/Thycotic/Thycotic.py

```diff
@@ -0,0 +1,127 @@
+    """
+    def __init__(self, server_url: str, username: str, password: str, proxy: bool):
+        super().__init__(base_url=server_url, proxy=proxy)
+        self._username = username
+        self._password = password
+        self._proxy = proxy
+        self._token = self._generate_token()
+        self._headers = {'Authorization': self._token, 'Content-Type': 'application/json'}
+
+    def _generate_token(self) -> str:
+        """Generate an Access token using the user name and password
+        :return: valid token
+        """
+        body = {
+            "username": self._username,
+            "password": self._password,
+            "grant_type": "password"
+        }
+
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        return "Bearer " + (self._http_request("POST", "/oauth2/token", headers=headers, data=body)).get('access_token')
+
+    def getToken(self):
+        return self._token
+
+    def getPasswordById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + secret_id + "/fields/password"
+        return self._http_request("GET", url_suffix)
+
+    def getUsernameById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + secret_id + "/fields/username"
+        return self._http_request("GET", url_suffix)
+
+
+def test_module(client: Client,) -> str:
+    if client._token == '':
+        raise Exception('Failed to get authorization token. Check you credential and access to Secret Server.')
+
+    return 'ok'
+
+
+def authenticate_token_command(client):
+    token = client.getToken()
+
+    return CommandResults(
+        readable_output=f'Access token for current session: {token}',
+        outputs_prefix='Thycotic.authenticate',
+        outputs_key_field='token',
+        raw_response=token,
+        outputs=token
+    )
+
+
+def secret_password_get_command(client, secret_id: str = ''):
+    secret_password = client.getPasswordById(secret_id)
+
+    return CommandResults(
+        readable_output=f"Retrieved password by ID {secret_id} {secret_password}",
+        outputs_prefix='Thycotic.secret',
+        outputs_key_field="secret_password",
+        raw_response=secret_password,
+        outputs=secret_password
+    )
+
+
+def secret_username_get_command(client, secret_id: str = ''):
+    secret_username = client.getUsernameById(secret_id)
+
+    return CommandResults(
+        readable_output=f"Retrieved username by ID {secret_id} {secret_username}",
+        outputs_prefix='Thycotic.secret',
+        outputs_key_field="secret_username",
+        raw_response=secret_username,
+        outputs=secret_username
+    )
+
+
+def main():
+    username = demisto.params().get('credentials').get('identifier')
+    password = demisto.params().get('credentials').get('password')
+
+    # get the service API url
+    url = demisto.params().get('url')
+    proxy = demisto.params().get('proxy', False)
+
```

Comment:
please add the insecure param, move it to the client and pass it to the super(), just as the proxy.
in the `BaseClient` it is called `verify` which is not insecure.
```suggestion
    proxy = demisto.params().get('proxy', False)
    verify = not demisto.params().get('insecure', False)
```